libbeat/docs/shared-ssl-config.asciidoc

```diff
@@ -52,6 +55,22 @@ heartbeat.monitors:
 -------------------------------------------------------------------------------
 endif::[]
 
+ifeval::["{beatname_lc}"=="metricbeat"]
+Example module with SSL enabled:
+
+[source,yaml]
+----
+- module: prometheus
```

Comment:
We could use just the `http` module here to reduce the noise around it but mention it works for all http based modules.

Reply:
@ruflin That makes sense. I looked at the docs for the http module, but it looks like we haven't updated the config to show the ssl options: https://www.elastic.co/guide/en/beats/metricbeat/master/metricbeat-module-http.html

Can you provide an example config? I can also update the [config.yml](https://github.com/elastic/beats/blob/master/metricbeat/module/http/_meta/config.yml) if you let me know what I need to add and where.

I guess I should also go through the other modules that have  `settings: ["ssl"]` to make sure that the config examples are updated?

Reply:
(After a quick spot check, I can see that there are several modules with the `ssl` flag that need to have their config examples updated).

Reply:
@jsoriano Could you chime in here? I remember you were updating recently quite a few things related to SSL.

Reply:
Oh yes, the thing is that we may need two different ssl documentation templates, one for HTTPS and another one for plain TLS, when adding the mongodb TLS support, the template added [something](https://github.com/elastic/beats/pull/7401/files#r197533811) about HTTP (I should have reviewed this :slightly_frowning_face: )

Reply:
@jsoriano I'm not quite sure how to proceed here. Please advise. Thanks!

Reply:
I think there should be two different templates, one for HTTP services and another one for TLS. Services that support HTTPS would use both, services that use TLS over TCP without HTTP would use the TLS one only.

In any case maybe I'm mixing things, I'm refering to the text that is added to modules documentation when they have `settings: ["ssl"]` in `fields.yml`, let's merge this and I'll open another PR with a proposal for the other thing.

Reply:
We talked about updating the example config, too. I should do this before we merge. Can someone provide an example?

Reply:
I think @ruflin refered to something like this as a generic example here:
```
- module: http
  namespace: "myservice"
  enabled: true
  period: 10s
  hosts: ["https://localhost"]
  path: "/stats"
  headers:
    Authorization: "Bearer test123"
  ssl.verification_mode: "none"
```

Not sure though if it'd be better to have a generic example or a real world example as the one with prometheus and the k8s apiserver.